src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1726,6 +1740,15 @@ protected override void InvokeDefaultAction(string path)
                                 return;
                             }
 
+                            //If symlink points to its parent directory then do not pursue symlink further
+                            if (!Platform.IsWindows && Platform.NonWindowsIsSymLink(recursiveDirectory))
+                            {
+                                if (DetectRecursiveSymLink(recursiveDirectory))
+                                {
+                                    recurse=false;
+                                    return;
+                                 }
+                            }
```

Comment:
I think we have the same problem in windows, for example:
```
 Directory of e:\psTest\test\foo

09/13/2016  04:26 PM    <DIR>          .
09/13/2016  04:26 PM    <DIR>          ..
09/13/2016  04:26 PM    <SYMLINKD>     foo [e:\psTest\test]
```
Can you please solve it for both win and unix?